src/main/java/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package strategy;
+
+import model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public int getQuantityToCalculate(FruitTransaction fruitTransaction) {
+        return -fruitTransaction.getQuantity();
+    }
```

Comment:
You need to add a check that ensures the balance is not negative after a purchase operation. If the balance is negative, you should throw a RuntimeException.